### PR TITLE
Add Union Support

### PR DESCRIPTION
**Pull request by abusi**
Now Union are supported and works as expected.

### CODE REVIEW

File: tartiflette/parser/nodes/field.py [hunk 1]
```diff
@@ -9,6 +9,20 @@
 from .node import Node
 
 
+def _get_raw_typename(raw):
+    try:
+        return raw["typename"]
```
Comment (tsunammis) [inline tartiflette/parser/nodes/field.py]: I would rather`_typename` instead.

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,42 +5,85 @@ All notable changes to this project will be documented in this file.
...
+
+#### Added
+
+- Support for Union and TypeCondition filtering
+
+#### Changed
+
+Now a resolver of a union type should explicitly define the value of `_typename` on it's return value.
+
+- If the resolver wants to return a `dict` a `_typename` key should be present.
```
Comment (tsunammis) [inline CHANGELOG.md]: `should` -> `must`

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,42 +5,85 @@ All notable changes to this project will be documented in this file.
...
+- If the resolver wants to return an `object` (a class instance), a `_typename` field should be present.
```
Comment (tsunammis) [inline CHANGELOG.md]: I wondering if it is called a `field` or `attribute`? What do you think?

`should` -> `must`

Comment (abusi) [reply CHANGELOG.md]: Yeah good catch, it's class attribute, I was thinking about graphql field. :)

Comment (abusi) [reply CHANGELOG.md]: but it stay should as if the class is correctly named, it will work.

**Pull request merged**